dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/HeaderParamEncoder.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import java.util.List;
+
+public interface HeaderParamEncoder<T> {
+    List<String> toHeaderParamValues(T value);
```

Comment:
actually, wouldn't this apply to query parameters and header parameters? It's paths that are slightly different because there are no lists or optionals.

Reply:
good point!

Reply:
Done